main.go

```diff
@@ -412,6 +421,20 @@ func getCACert() (string, error) {
 	return cs.Data["certificate"].(string), nil
 }
 
+func fetchOvpnKey(fqdn string) (string, error) {
+	path := strings.Join([]string{"secret", "data", strings.Trim(cfg.OvpnKey, "/")}, "/")
+	secret, err := client.Logical().Read(path)
+
+	if err != nil {
+		return "", err
+	}
+
+	if secret == nil {
+		return "", errors.New("Got no data from backend")
+	}
+	return secret.Data["data"].(map[string]interface {})["key"].(string), nil
```

Comment:
- Check `secret.Data` for being `nil` (might happen and cause a nil-pointer exception)
- Check `key` for being present in `secret.Data` (if not the `(string)` cast will fail)
- Remove `["data"]`, it will not be present there
- Apply `gofmt -s` to `main.go` OR remove the space in `interface {}`